owncloudApp/src/main/java/com/owncloud/android/workers/DownloadFileWorker.kt

```diff
@@ -0,0 +1,317 @@
+     * - Check that the file is not a folder
+     */
+    private fun areParametersValid(): Boolean {
+        val accountName = workerParameters.inputData.getString(KEY_PARAM_ACCOUNT)
+        val fileId = workerParameters.inputData.getLong(KEY_PARAM_FILE_ID, -1)
+
+        account = AccountUtils.getOwnCloudAccountByName(appContext, accountName) ?: return false
+        ocFile = getFileByIdUseCase.execute(GetFileByIdUseCase.Params(fileId)).getDataOrNull() ?: return false
+
+        return !ocFile.isFolder
+    }
+
+    /**
+     * Download the file or throw an exception if something goes wrong.
+     * We will initialize a listener to update the notification according to the download progress.
+     *
+     * File will be downloaded to a temporalFolder in the RemoteOperation.
+     * @see temporalFolderPath for the temporal location
+     */
+    private fun downloadFileToTemporalFile() {
+        downloadRemoteFileOperation = DownloadRemoteFileOperation(
+            ocFile.remotePath,
+            temporalFolderPath
+        ).also {
+            it.addDatatransferProgressListener(this)
+        }
+
+        // It will throw an exception if something goes wrong.
+        executeRemoteOperation {
+            downloadRemoteFileOperation.execute(client)
+        }
+    }
+
+    /**
+     * Move the temporal file to the final location.
+     * @see temporalFilePath for the temporal location
+     * @see finalLocationForFile for the final one
+     */
+    private fun moveTemporalFileToFinalLocation() {
+        val temporalLocation = File(temporalFilePath)
+
+        if (FileStorageUtils.getUsableSpace() < temporalLocation.length()) {
+            Timber.w("Not enough space to copy %s", temporalLocation.absolutePath)
+        }
+
+        val finalLocation = File(finalLocationForFile)
+        finalLocation.parentFile?.mkdirs()
+        val movedToTheFinalLocation = temporalLocation.renameTo(finalLocation)
+
+        if (!movedToTheFinalLocation) {
+            throw LocalStorageNotMovedException()
+        }
+    }
+
+    /**
+     * Update the database with latest details about this file.
+     *
+     * We will ask for thumbnails after a download
+     * We will update info about the file (modification timestamp and etag)
+     * We will update info about local storage (where it was stored and its size)
+     */
+    private fun updateDatabaseWithLatestInfoForThisFile() {
+        ocFile.apply {
+            needsToUpdateThumbnail = true
+            modificationTimestamp = downloadRemoteFileOperation.modificationTimestamp
+            etag = downloadRemoteFileOperation.etag
+            storagePath = finalLocationForFile
+            length = (File(finalLocationForFile).length())
+        }
+        saveFileOrFolderUseCase.execute(SaveFileOrFolderUseCase.Params(ocFile))
+
+        // To be done. Probably we will move it out from here.
+        //mStorageManager.triggerMediaScan(file.getStoragePath())
```

Comment:
Same here

Reply:
I didn't remove the comment because I will probably need it in upcoming PRs. But it won't be there when the final branch will be merged into master 👍